### PR TITLE
Summary output of RWIP

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

**Comment by joakim-hove** [comment]
Unless .... I'll merge this tomorrow.

**Comment by jokva** [comment]
No objections.

File: opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -286,6 +286,25 @@ quantity rpr(const fn_args& args) {
...
+    if (pair == args.regionCells.end())
+        return { 0.0 , measure::volume };
+
+    double RWIP = 0;
+    const std::vector<double>& swat = args.state[data::Solution::key::SWAT];
+    const auto& cells = pair->second;
+
+    for (auto cell_index : cells) {
+	size_t global_index = args.grid.getGlobalIndex( cell_index );
+        RWIP += swat[cell_index] * args.grid.getCellVolume( global_index );
```
Comment (pgdr) [inline opm/output/eclipse/Summary.cpp]: Should also multiply by `porv[cell_index]`

Comment (joakim-hove) [reply opm/output/eclipse/Summary.cpp]: > Should also multiply by porv[cell_index]

Well spotted  - thank you; amend pushed.

File: opm/output/eclipse/Summary.cpp [hunk 2]
```diff
@@ -286,6 +287,32 @@ quantity rpr(const fn_args& args) {
...
+
+
+    double RWIP = 0;
+    const std::vector<double>& swat = args.state[data::Solution::key::SWAT];
+    const auto& cells = pair->second;
+    const auto& properties = args.ecl_state->get3DProperties();
+    const auto& porv_data = properties.getDoubleGridProperty("PORV").getData();
+
+    for (auto cell_index : cells) {
+	size_t global_index = args.grid.getGlobalIndex( cell_index );
```
Comment (pgdr) [inline opm/output/eclipse/Summary.cpp:307]: somehow a tab has snuck in

**Comment by pgdr** [comment]
Ready to merge after the tab has been replaced.

**Comment by alfbr** [comment]
I am afraid you need to include the water formation volume factor, assuming that in place numbers are at surface conditions. For the Norne case, you need to calculate it from the water compressibility in the PVTW table. Since these numbers already are calculated in the simulator, and even done in parallel there, consider doing this on the simulator side.

**Comment by jokva** [comment]
We already compute original and current water/oil/gas in place in the simulator. Can we just pass this result as-is to output and just read that, or do I miss something? 

Line ~274 in SimulatorBase_Impl.hpp. If these numbers are the same we want to print it looks like we'll get ROIP, RGIP, RWIP, FWIP, FGIP and FOIP for free, which is nice.

**Comment by alfbr** [comment]
> We already compute original and current water/oil/gas in place in the simulator. Can we just pass this result as-is to output and just read that, or do I miss something?

I believe so, and that is what I am asking to consider.

File: opm/output/eclipse/Summary.cpp [hunk 3]
```diff
@@ -286,6 +287,32 @@ quantity rpr(const fn_args& args) {
...
+        RWIP += swat[cell_index] * porv_data[global_index];
```
Comment (andlaus) [inline opm/output/eclipse/Summary.cpp:308]: you probably also have to multiply this with B_W. (at least if RWIP is given in terms of surface volume, which IMO is the only sensible option.)

Also, I assume that "swat" corresponds to the water saturation given by the solution, not to the values given by the SWAT ECL property. (there is potential for confusion.)

**Review by andlaus — COMMENTED** [review]
IMO, this looks good once it has been clarified if the multiplication with B_W must be done or not.

Question: `R[OG]IP` are not yet supported? (I haven't folllowed this very closely.)

**Comment by andlaus** [comment]
> We already compute original and current water/oil/gas in place in the simulator. Can we just pass this result as-is to output and just read that, or do I miss something?

for the current `flow` in principle yes, in practice this probably implies some annoying opm-output API changes? maybe it is better to move the actual computation to the simulator and limit opm-output to a querying API for the requested quantities and to writing the results to disk.

**Comment by jokva** [comment]
> for the current flow in principle yes, in practice this probably implies some annoying opm-output API changes?

Possibly a change to the input (my guess would be `data::Solution`) structures, yes. No change unless you want to output fluid-in-place though.

> maybe it is better to move the actual computation to the simulator and limit opm-output to a querying API for the requested quantities and to writing the results to disk.

Isn't this what I suggested?

**Comment by andlaus** [comment]
> Isn't this what I suggested?

I suppose you did. it seems that I was just too stupid to notice, sorry...

**Comment by jokva** [comment]
No problems mate, just wanted to make sure I didn't miss a detail.

**Comment by joakim-hove** [comment]
> Since these numbers already are calculated in the simulator, and even done in parallel there, consider doing this on the simulator side.

We will do that; there are some unrelated refactorings in this patch I would like to salvage, but apart from that we will take this from the simulator.

**Comment by jokva** [comment]
> We will do that; there are some unrelated refactorings in this patch I would like to salvage, but apart from that we will take this from the simulator.

Alright, we'll take a look tomorrow then, yeah?

**Comment by joakim-hove** [comment]
OK - nothing worth keeping here. Closing.

**Pull request closed without merging**